storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/dataaccess/V4FinalizedKvStoreDao.java

```diff
@@ -270,6 +273,13 @@ public void addBlindedFinalizedBlock(
       addFinalizedBlockRootBySlot(block.getSlot(), root);
     }
 
+    @Override
+    public void addBlindedFinalizedBlockRaw(
+        final Bytes blockBytes, final Bytes32 root, final UInt64 slot) {
+      transaction.putRaw(schema.getColumnBlindedBlocksByRoot(), root, blockBytes);
+      addFinalizedBlockRootBySlot(slot, root);
+    }
```

Comment:
The risk with this function is we can throw anything in, so if it's used incorrectly it could definitely cause problems, but it does significantly improve the performance of block migration, as long as we don't also copy in blocks that need to be blinded (bellatrix)... I'd almost like for this to only be possible as part of the migration process, and outside of that it somehow is not accessible, but I'm not sure if that's just over-complicating things...

Reply:
I think this is fine to leave exposed. It's hard to imagine a case where we'd call this without being aware of the potential consequences.  Trying to make it available only to the migrator is going to add another level of generics and interfaces which just makes things harder to understand I think.